Cython/Compiler/Nodes.py

```diff
@@ -6689,9 +6695,19 @@ def analyse_expressions(self, env):
                     not (exc.args or (exc.arg_tuple is not None and exc.arg_tuple.args))):
                 exc = exc.function  # extract the exception type
             if exc.is_name and exc.entry.is_builtin:
+                from . import Symtab
                 self.builtin_exc_name = exc.name
                 if self.builtin_exc_name == 'MemoryError':
                     self.exc_type = None  # has a separate implementation
+                elif (self.builtin_exc_name == 'StopIteration' and
+                        isinstance(env, Symtab.LocalScope) and env.name == "__next__" and
```

Comment:
Is this `isinstance()` check needed? What else should there be when the parent is a C class scope?

Reply:
I could be wrong because I haven't tried it but potentially:

```
cdef class C
   class Nested:
      raise StopIteration
```
?

Reply:
Can we have an `is_local_scope` attribute then? That would seem more consistent with what we do pretty much everywhere else. (And it would avoid the local import.)